deepchem/splits/splitters.py

```diff
@@ -1181,6 +1180,143 @@ def _generate_scaffold(smiles: str, include_chirality: bool = False) -> str:
   return scaffold
 
 
+class FingerprintSplitter(Splitter):
+  """Class for doing data splits based on the Tanimoto similarity between ECFP4
```

Comment:
Not for this PR, but would there be a clear way to generalize this to non-Tanimoto fingerprints? That would help test the approach on non molecular datasets

Reply:
Most of the algorithm isn't even specific to molecules.  You just need some metric for similarity between samples.  Since this class works with molecules, it uses fingerprints.  For a different kind of data, you could use something else.